src/datatype/index.js

```diff
@@ -166,28 +200,29 @@ export class DataType extends TypedEmitter {
   /**
    * @param {string} docId
    * @param {{ lang?: string }} [opts]
+   * @returns {Promise<TDoc & { forks: string[] }>}
    */
   async getByDocId(docId, { lang } = {}) {
     await this.#dataStore.indexer.idle()
-    const result = /** @type {undefined | MapeoDoc} */ (
-      this.#sql.getByDocId.get({ docId })
-    )
+    const result = this.#sql.getByDocId.get({ docId })
     if (!result) throw new NotFoundError()
     return this.#translate(deNullify(result), { lang })
   }
 
   /**
    * @param {string} versionId
    * @param {{ lang?: string }} [opts]
+   * @returns {Promise<TDoc>}
    */
   async getByVersionId(versionId, { lang } = {}) {
     const result = await this.#dataStore.read(versionId)
-    return this.#translate(result, { lang })
+    return this.#translate(deNullify(result), { lang })
   }
 
   /**
-   * @param {MapeoDoc} doc
+   * @param {any} doc
```

Comment:
I was unable to figure out how to get this to type-check without `any`, much to my dismay.